mq/inmemory/mq.h

```diff
@@ -124,7 +131,17 @@ class MMQ final {
   // This method will be called from one thread, which is spawned and owned by an instance of MMQ.
   typedef CONSUMER T_CONSUMER;
 
-  // The only constructor requires the refence to the instance of the consumer of entries.
+  // Constructor that uses default `ProcessMessageConsumer`.
+  MMQ(size_t buffer_size = DEFAULT_BUFFER_SIZE)
+      : consumer_(*default_consumer_),
+        circular_buffer_size_(buffer_size),
+        circular_buffer_(circular_buffer_size_),
+        dropped_messages_count_(0u),
+        consumer_thread_(&MMQ::ConsumerThread, this) {
+    default_consumer_ = new mmq::ProcessMessageConsumer<T_MESSAGE>();
```

Comment:
Aaaaaa! `std::unique_ptr<> owner_default_consumer` and `.reset(new ...)` please!